System.IO.FileSystem/File.cs

```diff
@@ -54,49 +41,35 @@ public static class File
 
             if (string.IsNullOrEmpty(destFileName))
             {
-                throw new ArgumentException(nameof(sourceFileName));
+                throw new ArgumentException(nameof(destFileName));
             }
 
-            FileMode writerMode = overwrite ? FileMode.Create : FileMode.CreateNew;
-
-            FileStream reader = new FileStream(sourceFileName, FileMode.Open, FileAccess.Read);
-
-            try
+            if (sourceFileName == destFileName)
             {
-                using (FileStream writer = new FileStream(destFileName, writerMode, FileAccess.Write))
-                {
-                    long fileLength = reader.Length;
-                    //writer.SetLength(fileLength);
-
-                    byte[] buffer = new byte[_defaultCopyBufferSize];
+                throw new ArgumentException();

```

Comment:
Are you sure? This is not on the .NET docs... maybe this will just return.... please double check.

Reply:
Good catch.